miden-tx/src/executor/mod.rs

```diff
@@ -141,10 +141,9 @@ impl<D: DataStore> TransactionExecutor<D> {
         account_id: AccountId,
         block_ref: u32,
         notes: &[NoteId],
-        tx_args: Option<TransactionArgs>,
+        tx_args: TransactionArgs,
```

Comment:
Are we making `tx_args` non-optional because transactions with no tx args are expected to be rather rare? If so, maybe we should add something like `TransactionArgs::empty()` constructor. Though, I guess one can always use `TransactionArgs::default()` to create empty transaction args.

Also, this change will likely to break the node and the client - so, if we keep it like this, could you make a corresponding change in `miden-node` and cc @igamigo for the client.